Python/ibmcloudsql/SQLQuery.py

```diff
@@ -311,57 +595,198 @@ def get_result(self, jobId, pagenumber=None):
+            elif job_details['status'] != 'completed':
+                raise ValueError('SQL job with jobId {} did not finish successfully. No result available.')
+
+            if "resultset_location" not in job_details:
+                return None
+            result_location = job_details['resultset_location']
+            url_parsed = self.analyze_cos_url(result_location)
+            result_bucket = url_parsed.bucket
+            result_endpoint = url_parsed.endpoint
+            result_objects_df =  self.list_cos_objects(job_details['resultset_location'])
+            result_objects_df['Bucket'] = result_bucket
+            result_objects_df['ObjectURL'] = result_objects_df.apply(lambda x: 'cos://%s/%s/%s' % (result_endpoint, result_bucket, x['Object']), axis=1)
+            return result_objects_df
+
+        job_id = jobId
+        if blocking:
+            job_running = True
+            while job_running:
+                try:
+                    x = list_results(job_id)
+                    job_running = False
+                except ValueError as e:
+                    if "running" in str(e):
+                        pass
+                    else:
+                        raise e
+        else:
+            x = list_results(job_id)
+        return x
+
+    def rename_exact_result(self, jobId, blocking=True):
+        """
+        A SQL Query can store data into partitioned/paginated multiple objects, or single object.
+
+        Even with single object, indeed, multiple objects are created, two of them has size 0.
+        (<URL>/_SUCCESS, and <URL>/) beside the ones that hold data (<URL>/<data1>, <URL>/<data2>)
+
+        This API delete the two 0-size objects, and keep only the one that holds data.
+
+        Parameters
+        ----------
+        job_id : str
+            A string representation of job_id
 
-        result_location = job_details['resultset_location']
-        url_parsed = self.ParsedUrl(result_location)
-        result_bucket = url_parsed.bucket
-        result_endpoint = url_parsed.endpoint
-        result_objects_df =  self.list_cos_objects(job_details['resultset_location'])
-        result_objects_df['Bucket'] = result_bucket
-        result_objects_df['ObjectURL'] = result_objects_df.apply(lambda x: 'cos://%s/%s/%s' % (result_endpoint, result_bucket, x['Object']), axis=1)
-        return result_objects_df
+        blocking: bool, optional
+            Default is wait
 
-    def rename_exact_result(self, jobId):
+        Returns
+        --------
+        None
+
+        Raises
+        ------
+        ValueError
+            If the job_id is the job in that the result is "PARTITIONED BY" or "pagesize=" or "PARITIONED EVERY x ROWS" is used
+            or the rename_exact_result() has been applied to this job_id.
+        """
         self.logon()
 
         job_details = self.get_job(jobId)
         job_status = job_details.get('status')
+
+        while job_status == "running" and blocking is True:
```

Comment:
We already have the explicit method wait_for_job that the user knows and can use to ensure job termination before working with the results. So change the parameter from "blocking=True" to "wait=False" and describe it also in the way that passing in True it is equivalent to running wait_for_job beforehand. And of course simply call wait_for_job when True is passed in at the start of this method.

Reply:
Done in commit `1e12447646484b05db50f3f7923384379cede00`